hw/ip/aes/doc/_index.md

```diff
@@ -133,14 +139,25 @@ The AES unit operates as follows:
 1. The initial key and configuration is provided to the AES unit via a set of control and status registers (CSRs) accessible by the processor via TL-UL bus interface.
    Each key register has to be written at least once.
    The order in which the registers are written does not matter.
+1. The processor provides the initialization vector (IV) or initial counter value to the four IV registers via TL-UL bus interface in CBC or CTR mode, respectively.
+   Each IV register has to be written at least once.
```

Comment:
what happens if the processor doesn't write the IV, or all 4 IV? Can't it in theory
just leave the value from the previous execution? What happens if not all registers
are written, literally from a HW point of view?

Reply:
> what happens if the processor doesn't write the IV, or all 4 IV? Can't it in theory
> just leave the value from the previous execution? 
In CBC and CTR mode, the IV register is always updated by the AES unit itself after the IV value has been consumed. This way we never use the exact same IV value twice. Now, if you want to encrypt a new message (not just another block of the previous message) without writing a new IV, the IV for the next block of the previous message is used.

> What happens if not all registers are written, literally from a HW point of view?
In this case the AES unit will not start. My reasoning was that we don't want someone to mess with partially updated IVs. Just like the actual input data (in automatic mode every input data register needs to be written at least once). In manual mode, we don't care about partially updated IVs and input data, nor about unread output data.

I have tried to clarify this in the documentation.

Reply:
sg